GOVERNANCE.md

```diff
@@ -0,0 +1,458 @@
+- Making sure that contributors know how decisions are made and who they can address concerns/questions to
+- Making sure that team members know what is required of them for the project
+
+
+## The project vision
+
+In the [vision.md](./vision.md) the goal of the project is encoded.
+It specifies the different functional requirements and non-functional requirements of the project as a whole.
+All the stakeholders must agree on the vision upon joining.
+
+- The current stakeholders are: Software AG, and IFM
+
+Changes to the vision need to be done in a plenum of stakeholders, this makes sure that everyone is aware and agrees to the evolution of the project.
+
+
+
+## Team Structure
+
+* Teams are either single contributors or a group of them
+* They are responsible for a specific part of project, and as such are the maintainers of those parts
+    * This includes: Reviewing pull requests, triaging issues in their assigned area, and general maintenance
+* Each team is assigned a single or multiple subtrees of the project, as defined by the project structure
+* The teams follow the project hierarchy for decisions, i.e. higher decisions take precedence
+
+The important bits:
+
+At the 'top' are the maintainers, whose job is to define and realize the project vision of thin-edge.io.
+Maintainers should strive for agreement based on the following factors:
+
+- The vision of the whole project
+- Feasibility, accounting for future technical and cultural shifts
+- The health of the project going forward
+
+Overall, the maintainers should not be beholden to their respective companies, and instead to the health of the project and its community.
+
+Underneath the maintainers are the teams or individuals, that each are allowed to  make similar decisions about the project *in the area they have been delegated*.
+
+
+- Some decisions are global in the project, like the core/API
+    - Team members should try to make sure to explore possible options together before calling for time-consuming solutions like a vote. This mostly includes uncontroversial changes that are low impact or have already been agreed upon
+    - If an exclusive choice has to be made (where it is not possible to entertain two conflicting approaches), and no clear side is more advantageous to pick, a vote should be held with the majority opinion being the final decision.
+
+- Some decisions are local, e.g. a plugin does not impact others with its behaviour, however it still needs to be a 'good citizen' in the project (CI, format, etc..)
+    - These should be clearly scoped in their respective project part
+    - However, if needed a 'higher up' decision can be requested if no consensus is achieved
+
+
+
+## Project Structure
+
+The project has a hierarchy to solve these two problems:
+
+1. How can users make sure that their voice and interests are represented in the project?
+    - Each member party appoints one core team member
```

Comment:
This ratio needs to also account for investment made by member party.

Reply:
Investment into what? I consider this to be a particularly bad choice for OSS projects:

- Who gets to decide which investment counts? If its self-reported, then it might as well just be made up. If there is some need for proof, then it will preclude any able but just not rich people from feeling like they have a say (but they absolutely should have a say)
- Money is not necessarily a good indicator for aptitude or competence in leading a project. While several OSS projects do have 'paid' core membership positions, the size of the investment is not relevant. If anything, the knowledge and experience of the individual core members is what ultimately put them in such a position.
- Software AG and IFM have already agreed to start as equal partners. This would assuredly mean this balance gets broken in the future.

Reply:
@itsroshan  @TheNeikos  I am not sure to what investement we are referring to but here are my thoughts on this topic: 
- I believe the simplification of "Each member party appoints one core team member" is not exact enough , do we refer to the fact each member party appoints one maintainer?
- And what is "core team" ? Is the the maintainer group? 

- If we talk about maintainers: 
- Maintainers should be people who: share the same vision, interact with the community, are embracing open decision making and documentation, the mainttainer role should not be given away based on the fact that someone belongs to a partner company but instead based on substantial contribution to the project and/or extensive support of the community and/or an other valuable activity - so according to a meritocracy anyone who has exceptionally contributed in one of such aspects should "earn the right" to become a maintainer . 
- We can start with a small group e.g. 2 maintaners and from there use the vote to nominate new members after they have proven themselves
- It will be "natural" that if a company will support the project with "investment" (of any kind) certain people will be able to spend much time on contributions, community work etc and for one of those areas would increase the chance to become a maintainer  (but this is not exclusive to members of the contributing partner companies)
- Next to the 2 maintainers we might introduce a "steering group or board" which consists of the company decision makers from the partner companies to ensure big decisions are in line with the companies interests in the project

Reply:
Follow-up task: https://github.com/thin-edge/thin-edge.io/issues/1377